pkg/pkgrepository/reconcile_timer_test.go

```diff
@@ -193,3 +193,33 @@ func TestFailedIsReadyAt(t *testing.T) {
 		t.Fatalf("Expected app to not be ready under syncPeriod of 2s")
 	}
 }
+
+func TestIsReadyAtWithStaleDeployTime(t *testing.T) {
+	syncPeriod := 2 * time.Second
+	timeNow := time.Now()
+	timeOfReady := timeNow.Add(syncPeriod)
+
+	app := v1alpha1.App{
+		Spec: v1alpha1.AppSpec{
+			SyncPeriod: &metav1.Duration{Duration: syncPeriod},
+		},
+		Status: v1alpha1.AppStatus{
+			Fetch: &v1alpha1.AppStatusFetch{
+				UpdatedAt: metav1.Time{Time: timeOfReady},
+				Error:     "I've failed you",
+			},
+			Deploy: &v1alpha1.AppStatusDeploy{
+				UpdatedAt: metav1.Time{Time: timeNow},
+			},
+			ConsecutiveReconcileFailures: 1,
+			GenericStatus: v1alpha1.GenericStatus{
+				Conditions: []v1alpha1.AppCondition{{Type: v1alpha1.ReconcileFailed}},
+			},
+		},
+	}
+
+	isReady := NewReconcileTimer(app).IsReadyAt(timeOfReady.Add(1 * time.Second))
+	if isReady {
+		t.Fatalf("Expected app not to be ready, because deploy time is stale")
```

Comment:
(same),   and maybe an obligatory "have we already weighed the costs/benefits of DRY-ing out the reconcile timers?"

Reply:
Agreed, I think DRYing these up would be good, but it should probably be a piece of a lager effort to DRY up the entire package repository impl (its currently a copy and paste of the app package).

Reply:
intent to keep them separate (we copy-pasta them intentionally) because we should be moving towards package repository be decoupled from CRs-as-storage so current implementation is temporary.